languages/ruby/exercises/concept/instance-variables/.docs/introduction.md

```diff
@@ -0,0 +1,56 @@
+## Object state, instance variables
+
+Objects can hold their own state by setting _instance variables_. _Instance variables_ are created a prefixing `@` to the variable name.
+
+```ruby
+@instance_variable = 2
+```
+
+This is often done by defining an `initialize` method to set an initial state.
+
+```ruby
+class Airplane
+  def initialize
+    @wings = 2
+  end
+end
+```
+
+The `initialize` method may also take arguments, so that each instance can start with a custom state:
+
+```ruby
+class Suitcase
+  def initialize(locked)
+    @locked = locked
+  end
+end
+```
+
+Consider _instance variables_ to be private from external read and writes. _Class instance methods_ should be used for getting and setting instance variables:
+
+```ruby
+class Suitcase
+  #...
+
+  def locked? # Methods returning true or false should have trailing `?`
+    @locked
+  end
+
+  def unlock! # Methods which mutate state should have trailing `!`
```

Comment:
Unless a method name is obvious that it mutates state, the last character of a method that changes state will have a `!` as the last character.

For example:  A method named `delete` does not need the `!` character, but a method named `color!` would indicate that the color would change.

Reply:
This was resolved, but Ruby convention is that the bang is unnecessary to state mutation, unless it is unclear.  Unlocking something is clearly a mutating behavior.  You can not, for example, unlock a door without changing its state, it should be intuitive that unlocking a pass a pass would mutate the pass.